src/libcadet/model/parts/TractorConvectionDispersionOperator.cpp

```diff
@@ -0,0 +1,1526 @@
+						if (wantJac)
+						{
+							_jacC.centered(offsetCur, 0) += static_cast<double>(finalFactor);
+							_jacC.centered(offsetCur, _nComp) -= static_cast<double>(finalFactor);
+						}
+					}
+				}
+			}
+		}
+	}
+
+	if (cadet_unlikely(_nRad <= 1))
+		return 0;
+
+
+    // NOTE: Tractor flux implementation
+	for (unsigned int col = 0; col < _nCol; ++col)
+	{
+		const unsigned int offsetColBlock = col * _nRad * _nComp;
+		ResidualType* const resColBlock = res + offsetC + offsetColBlock;
+		StateType const* const yColBlock = y + offsetC + offsetColBlock;
+
+		for (unsigned int rad_orig = 0; rad_orig < _nRad; ++rad_orig)
+		{
+			const unsigned int offsetToRadOrigBlock = rad_orig * _nComp;
+			const unsigned int offsetColRadOrigBlock = offsetColBlock + offsetToRadOrigBlock;
+			ResidualType* const resColRadOrigBlock = resColBlock + offsetToRadOrigBlock;
+			StateType const* const yColRadOrigBlock = yColBlock + offsetToRadOrigBlock;
+
+            /* for (unsigned int rad_dest = 0; rad_dest < (_nRad - rad_orig); ++rad_dest) */
+            for (unsigned int rad_dest = 0; rad_dest < _nRad; ++rad_dest)
+            {
+                const unsigned int offsetToRadDestBlock = rad_dest * _nComp;
+                const unsigned int offsetColRadDestBlock = offsetColBlock + offsetToRadDestBlock;
+                ResidualType* const resColRadDestBlock = resColBlock + offsetToRadDestBlock;
+                StateType const* const yColRadDestBlock = yColBlock + offsetToRadDestBlock;
+
+                for (unsigned int comp = 0; comp < _nComp; ++comp)
+                {
+                    const unsigned int offsetCur_orig = offsetColRadOrigBlock + comp;
+                    const unsigned int offsetCur_dest = offsetColRadDestBlock + comp;
+                    StateType const* const yCur_orig = yColRadOrigBlock + comp;
+                    StateType const* const yCur_dest = yColRadDestBlock + comp;
+                    ResidualType* const resCur_orig = resColRadOrigBlock + comp;
+                    ResidualType* const resCur_dest = resColRadDestBlock + comp;
+
+                    // NOTE: Tractor implementation follows
+					const ParamType exchange_orig_dest_comp = static_cast<ParamType>(_exchangeMatrix[rad_orig * _nRad * _nComp + rad_dest * _nComp + comp]);
+					if (cadet_likely(exchange_orig_dest_comp > 0.0))
+					{
+						*resCur_orig += exchange_orig_dest_comp * yCur_orig[0];
+						*resCur_dest -= exchange_orig_dest_comp * yCur_orig[0];
+
+						// TODO: @Sam, please check this
+						if (wantJac)
+						{
+							_jacC.centered(offsetCur_orig, 0) += static_cast<double>(exchange_orig_dest_comp);
+							_jacC.centered(offsetCur_orig, offsetCur_dest) -= static_cast<double>(exchange_orig_dest_comp);
+							_jacC.centered(offsetCur_dest, 0) -= static_cast<double>(exchange_orig_dest_comp);
+							_jacC.centered(offsetCur_dest, offsetCur_orig) += static_cast<double>(exchange_orig_dest_comp);
```

Comment:
We have changes in two different residuals (is this correct? Could already be accounted for in `_exchangeMatrix`). In each change, only one state vector item is involved. Hence, we should see two changes in the Jacobian:
* `d resCur_orig / d yCur_orig = exchange_orig_dest_comp`
* `d resCur_dest / d yCur_orig = -exchange_orig_dest_comp`

We also need to take into account that the second argument of `centered()` refers to the offset on the main diagonal of the matrix. That is, `centered(x,y)` refers to matrix item `mat_{x, x+y}`.

This leads to the code
```cpp
_jacC.centered(offsetCur_orig, 0) += static_cast<double>(exchange_orig_dest_comp);
_jacC.centered(offsetCur_dest, static_cast<int>(offsetCur_orig) - static_cast<int>(offsetCur_dest)) -= static_cast<double>(exchange_orig_dest_comp);
```
Since `resCur_orig` corresponds to `yCur_orig`, we get diagonal `0`. For the second, we need the offset from `resCur_dest` to `yCur_orig`. From our current position (diagonal `0`, equivalent to `resCur_dest`), we go back to the origin (`-offsetCur_dest`) and from there to `yCur_orig` (`+offsetCur_orig`). Note that we need to convert to `int` because `unsigned int` does not allow for negative offsets.